core/node/fee_model/src/l1_gas_price/main_node_fetcher.rs

```diff
@@ -73,6 +99,27 @@ impl MainNodeFeeParamsFetcher {
 
 impl BatchFeeModelInputProvider for MainNodeFeeParamsFetcher {
     fn get_fee_model_params(&self) -> FeeParams {
-        *self.main_node_fee_params.read().unwrap()
+        let fee_params = *self.main_node_fee_params.read().unwrap();
+        let batch_fee_input = *self.main_node_batch_fee_input.read().unwrap();
+        if batch_fee_input.is_none() {
+            return fee_params;
+        }
+        let batch_fee_input = batch_fee_input.unwrap();
+        match fee_params {
+            FeeParams::V1(..) => FeeParams::V1(FeeParamsV1 {
+                config: FeeModelConfigV1 {
+                    minimal_l2_gas_price: batch_fee_input.fair_l2_gas_price(),
+                },
+                l1_gas_price: batch_fee_input.l1_gas_price(),
+            }),
+            FeeParams::V2(params) => {
+                return FeeParams::V2(FeeParamsV2::new(
+                    params.config(),
+                    batch_fee_input.l1_gas_price(),
+                    batch_fee_input.fair_pubdata_price(),
+                    params.conversion_ratio(),
+                ));
+            }
+        }
```

Comment:
Why do we need all of this logic? Can't we just override `get_batch_fee_input_scaled` instead and return `*self.main_node_batch_fee_input.read().unwrap()` there? EN's scaling factor will be ignored but arguably this logic is not designed well in general (i.e. is there a reason to scale gas price differently on EN?). Also, `fn get_fee_model_params(&self) -> FeeParams` being a part of the base trait is strange too, it's more like an implementation detail for half of the implementations.